### PR TITLE
SRCH-4772 Remove Azure

### PR DESCRIPTION
**Pull request by lorawoodford**
## Summary
- Full removal of the long-deprecated Azure search engine, and related code and tests;
- Remaining instances of the Azure-related module codes (`AIMAG` `AWEB` `AZCI` `AZCW`) will come in a subsequent PR;
- BingV5 will be similarly removed in an additional PR;
- Note: A large portion of this PR includes re-recorded or permanently deleted VCR cassettes for both rspec and feature specs, because:
     - Since we're removing a fair chunk of code, I wanted to make sure we caught and removed all unnecessary cassettes at the same time; 
     - During the course of the work on this ticket, I discovered that most of our cassettes were recording with the old, deprecated, and no longer functioning Bing cognitive API, so wanted to re-record them (for why I didn't/couldn't record BingV7 Image searches successfully, see Slack);
     - I also noticed while working through these cassettes that I had improperly set us to globally record exact bytes, which is definitely not necessary and makes our cassettes difficult to deal with.  I've [modified that](https://github.com/GSA/search-gov/pull/1413/files#diff-d0368ac40b9e2814bd2678f957b9b9bbee394be126985b5cf2780b06ee2899f0), so the re-recorded cassettes now contain human-readable responses;
     - Finally, I just wanted to fully confirm old cassettes weren't masking new problems with any existing search behaviors introduced by this removal.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [ ] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
       ⚠️ 3 pre-existing CodeClimate failures popped stemming from files unchanged as part of this PR.  I suspect this is because I rerecorded some cassettes as part of this work.  Long story short, these are not new findings, all 3 stem from long-existing views that are not modified in this PR, and all 3 will be themselves deleted when the "old" SERP is removed in a few months.  Seeing as these would be nontrivial to fix, I'm requesting we ignore these failures (CC passing is an optional branch protection), and proceed as normal.
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: lib/api/v2/search_as_json.rb [hunk 1]
```diff
@@ -33,17 +33,6 @@ def as_json_results_to_hash
     @results.collect { |result| as_json_result_hash result }
```
Comment (krbhavith) [inline lib/api/v2/search_as_json.rb:33]: Is this `as_json_result_hash` calling the method in `Api::V2::NonCommercialSearch`?

Comment (lorawoodford) [reply lib/api/v2/search_as_json.rb]: @krbhavith Yes, good question.  I've moved the long remaining `as_json_result_hash` method from `NonCommercialSearch` to this module to make that fact even more obvious.  `CommercialSearch` itself is essentially deprecated as well, but still pops up in places related to the deprecated `ApiDocs` search functionality.  This was already a huge PR so I didn't want to take it any further, but I'm going to confirm there are still investigation tickets around covering the future removal of all that code as well (at which point this should become even simpler).  Thanks.

**Comment by lorawoodford** [comment]
@krbhavith Just another ping on this.  I know it's a lot of changes, so definitely feel free to reassign/ping slack for help if you're swamped.

**Review by krbhavith — APPROVED** [review]
LGTM

**Pull request merged**